### PR TITLE
Support TypeScript 5

### PR DESCRIPTION
**Pull request by fisker**
## Description

https://github.com/microsoft/TypeScript/pull/51387

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by jakebailey** [comment]
Not sure if you saw it in my PR, but https://github.com/jakebailey/prettier-style-ts-tree-shaking is my PoC for getting a similar result for 5.0.

**Comment by fisker** [comment]
I did. But I'm not going to use rollup.

**Comment by jakebailey** [comment]
Oh, I see, you're using esbuild's comments like `// src/compiler/utilities.ts`. That's a really clever idea.

**Comment by fisker** [comment]
Let's remove this https://github.com/prettier/prettier/blob/3fc6298c0ea633cc596b234b49869afa86689ba5/src/language-js/print/class.js#L307

**Comment by fisker** [comment]
Changlog missing.

File: tests/format/typescript/decorator-auto-accessors/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -1,5 +1,65 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`decorator-auto-accessor-like-property-name.ts format 1`] = `
+====================================options=====================================
+parsers: ["typescript"]
+printWidth: 80
```
Comment (fisker) [inline tests/format/typescript/decorator-auto-accessors/__snapshots__/jsfmt.spec.js.snap:6]: Those tests need run in --no-semi mode, let's move to a separate dir and test.

**Comment by fisker** [comment]
https://github.com/nodejs/node/issues/47096

**Pull request merged**